staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/kcp_rest_options_getter.go

```diff
@@ -0,0 +1,34 @@
+package apiserver
+
+import (
+	"fmt"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/registry/generic"
+)
+
+type apiBindingAwareCRDRESTOptionsGetter struct {
+	delegate generic.RESTOptionsGetter
+	crd      *apiextensionsv1.CustomResourceDefinition
+}
+
+func (t apiBindingAwareCRDRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource) (generic.RESTOptions, error) {
```

Comment:
Note, any error we return from this function will **panic** the apiserver (https://github.com/kcp-dev/kubernetes/blob/feature-logical-clusters-1.23/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go#L108)